parsers/osx_file_parser.py

```diff
@@ -53,3 +54,68 @@ def Parse(self, cmd, args, stdout, stderr, return_val, time_taken,
     serial_number = hardware_list["serial_number"]
 
     yield rdfvalue.HardwareInfo(serial_number=serial_number)
+
+class OSXLaunchdPlistParser(parsers.FileParser):
+  """Parse Launchd plist files into LaunchdPlist objects."""
+
+  output_types = ["LaunchdPlist"]
+  supported_artifacts = ["OSXLaunchAgents", "OSXLaunchDaemons"]
+
+  def Parse(self, stat, file_object, knowledge_base):
+    """Parse the History file."""
+    _, _ = stat, knowledge_base
+
+    plist = binplist.readPlist(file_object)
+
+    try:
+      username = plist["UserName"]
```

Comment:
Ah, I tried this and it complained that I can't use a dict.  I was missing the ** though so that explains that.  I will probably change the protobuf names to just be lower and not add the underscores I had and then use this method for any of the terms I can pull out directly without any manipulation.  Thanks!